AngularPart1/Factories/requestFactory.js

```diff
@@ -0,0 +1,47 @@
+app.factory("localRequestFactory",['$resource', function ($resource) {
+    var taskList = [];
+    return {
+       /* getResource:function(todo){
+            return $resource('', todo,
+                {
+                    'query' : { method:'GET', url:'http://localhost:3000/todos'},
+                    'update': { method:'PUT', url:'http://localhost:3000/todos/:id',params:{id:todo.date}},
+                    'add': { method:'POST', url:'http://localhost:3000/todos',params:todo },
+                    'delete': { method:'DELETE', url:'http://localhost:3000/todos/:id',params:{id:todo._id} }
+                });
+        },*/
+        getTasks: function getTasks(callback) {
+           /* this.getResource().query().
+            $promise.then(function(tasks) {
+                callback(tasks)
+            });*/
+            var resource = $resource('http://localhost:3000/todos');
+            resource.query()
+                .$promise.then(function(tasks) {
+                    callback(tasks)
+            });
+        },
+        addTask: function addTask(todo) {
+            //this.getResource().add(todo);
+            var Todo = $resource('http://localhost:3000/todos', {}, {
+                create: { method: 'POST', url: 'http://localhost:3000/todos', params: todo }
+            });
+            Todo.create(todo);
+        },
+        updateTask: function (todo) {
+           // this.getResource().update(todo);
+            var Todo = $resource('http://localhost:3000/todos/'+todo.date, {}, {
+                update: { method: 'PUT', url: 'http://localhost:3000/todos/'+todo.date, params: todo }
+            });
+            Todo.update(todo);
+        },
+        removeTask: function removeTask(todo) {
+            //this.getResource().delete(todo);
+            var Todo = $resource('http://localhost:3000/todos/'+todo._id, {}, {
```

Comment:
I would expect describing resource only once and reusing it as a class. Please see examples in official documentation (CreaditCard): https://docs.angularjs.org/api/ngResource/service/$resource